openedx/openedx-platform
# [FC-0049] Handle tags when importing/exporting courses

## Description

- Export: Add `tags.csv` when exporting a course. 
- Import: Read `tags.csv` and creates object_tags depending on the [scenario](https://github.com/openedx/modular-learning/issues/182#issue-2086188200)
-  Changes to use `object_tag._export_id`
- Update export/import tags separator to `;`

## Supporting information

- Github issue, closes https://github.com/openedx/modular-learning/issues/182
- Depends on: https://github.com/openedx/openedx-learning/pull/172
- Internal ticket: [FAL-3624](https://tasks.opencraft.com/browse/FAL-3624)

## Testing instructions

- Run `make requirements` on cms shell.
- Run `make migrate` on cms shell.
- Make sure you have Taxonomy/Tags data setup locally and verify that the command runs well: https://github.com/open-craft/taxonomy-sample-data/
- Create a new/empty course.
- Export empty course. Verify that `tags.csv` is not included in the export files.
- Export a Sample Taxonomy Course. Verify that `tags.csv` is included in the export files. Open and verify the content of the file.

**Scenario 1:**

- Import the new generated export files into the empty course. Verify that the tags are visible.

**Scenario 2:**

- Open `tags.csv` and add tags that doesn't exist on a taxonomy (FlatTaxonomy in this example). Save.
- Import with the edited `tags.csv`. Verify that the new tags are not visible.
- Go to [Taxonomy list page](http://localhost:2001/taxonomies/). Export FlatTaxonomy as csv and add the new tags. Import.
- Verify that the new tag is visible on the course.

**Scenario 3:**

- Open `tags.csv` and add tags that doesn't exist on a taxonomy that doesn't exist. You need to add a new column with a new `export_id`. Save.
- Import with the edited `tags.csv`. Verify that the new tags are not visible.
- Create a csv file to import a new taxonomy with the new tags
- Go to [Taxonomy list page](http://localhost:2001/taxonomies/). Import the new taxonomy with the same `export_id` added on `tags.csv`.
- Open `Manage organizations` menu of the new taxonomy and add it to all organizations.
- Verify that the new taxonomy and tags are visible on the course.

**Scenario 4:** https://github.com/openedx/modular-learning/issues/204

## Other information

- [x] Merge and bump version: https://github.com/openedx/openedx-learning/pull/172
